### PR TITLE
feature/add-ui-support: add UI support to starter

### PR DESCRIPTION
**Pull request by mh3720**
This is all Robert's fault.

Do NOT merge.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -7,6 +7,10 @@ before_install:
   - cp .env.example .env
   # Stop the postgres instance travis sets up automatically
   - sudo /etc/init.d/postgresql stop
+  # Build assets
+  - npm install
+  - npm install -g gulp
+  - gulp --production
```
Comment (bkuhl) [inline .travis.yml]: This should be done in the Dockerfile, otherwise the built assets won't be deployed within the container.

**Comment by bkuhl** [comment]
Ok... Just an FYI that I'm pretty strongly opinionated that this should remain all raw Laravel and should contain _no_ realpage-specific practices or code.

File: .travis.yml (hunk 1 above)
Comment (RPbarfield) [inline .travis.yml]: We only want this to run in containers that are going to be deployed. Locally, we will just run a standard "gulp". It is working beautifully for the RentJoy application deployment.

**Comment by bkuhl** [comment]
Currently this repo is open source and is open for anyone to download.  It's unmodified Laravel just with some added components to enhance development.  This PR introduces the first realpage-specific logic into this starter.  I don't think this should be merged into the open source version of the `laravel-starter`.

I think we should close this PR and create a private fork of the starter and re-open this PR there.  Thoughts?

File: .travis.yml (hunk 1 above)
Comment (bkuhl) [inline .travis.yml]: There's no `docker commit` or anything following these commands so how are the gulp-generated files staying on the container?

Comment (RPbarfield) [inline .travis.yml]: It runs these commands before the files in the repo are added to the container.

**Comment by RPbarfield** [comment]
Not sure why we are concerned about making this repo RealPage specific. I'm more concerned about this repo helping our team/company get to the finish line faster. The open source community has other options if they don't like the direction we are going.

**Comment by bkuhl** [comment]
We have the same objective.  I don't want to see us put teams in a box where this repo becomes too specialized/opinionated about how things are implemented.

File: gulpfile.js [hunk 2]
```diff
@@ -13,7 +17,23 @@ require('laravel-elixir-vue');
...
+        .sass(['core.scss','media-query.scss'],'./public/assets/css/custom.css')
+        .scripts([
+            './resources/assets/foundation-ui-library/get-latest-from-repo.js'
+        ],'./public/assets/js/')
+        .jshint('./resources/assets/js/*.js')
+        .version(['./public/assets/css','./public/assets/js'])
+        .copy([
+            './resources/assets/fonts'
+        ],'./public/build/assets/fonts')
+        .copy('./resources/assets/images','./public/build/assets/images')
```
Comment (bkuhl) [inline gulpfile.js]: These assets need to be pulled in via dependency management solutions (npm install).

**Comment by bkuhl** [comment]
Closing this since there's a project-specific starter in house.

**Pull request closed without merging**